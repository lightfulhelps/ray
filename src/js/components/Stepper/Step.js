// @flow
import * as React from 'react';
import Col from '../Col/Col';
// import stepper from '../../../scss/custom/stepper';

const Step = props => {
  const { label, value, isLast } = props;
  console.log('props', props)
  return (
    <Col>
      <div className="d-flex align-items-center">
        <div className="step-value border border-black rounded-circle d-flex align-items-center justify-content-center mr-1">
          {' '}
          {value}{' '}
        </div>
        <span className="mr-1">{label}</span>
        {!isLast && <div className="line" />}
      </div>
    </Col>
  );
};

export default Step;
