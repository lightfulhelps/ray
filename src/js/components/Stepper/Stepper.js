// @flow
import * as React from 'react';
import Step from './Step';
import Row from '../Row/Row';

const Stepper = props => {
  const { steps, activeStep } = props;
  return (
    <Row>
      {steps.map((step, i) => (
        <Step
          key={i}
          label={step.label}
          value={step.value}
          isLast={i === steps.length - 1}
          activeStep={activeStep}
          thisStep={i + 1}
        />
      ))}
    </Row>
  );
};

export default Stepper;
