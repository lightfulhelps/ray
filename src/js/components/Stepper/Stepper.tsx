import * as React from 'react';
import classNames from 'classnames';
import Step from '../Step/Step';

type StepType = {
  label: string;
  onClick?: (event: React.MouseEvent) => void;
  value: number | string;
};

type Props = {
  activeStep: number;
  className?: string;
  steps: Array<StepType>;
  [key: string]: any;
};

const Stepper: React.FC<Props> = ({ activeStep, className, steps, ...other }: Props) => {
  const classes = classNames(
    className,
    'stepper',
    'd-flex flex-column flex-sm-row justify-content-between'
  );

  if (!Array.isArray(steps) || !activeStep) {
    return null;
  }

  return (
    <div {...other} className={classes} data-test-id="stepper">
      {steps.map((step, i) => (
        <Step
          activeStep={activeStep}
          isLast={i === steps.length - 1}
          key={i}
          onClick={step.onClick}
          label={step.label}
          thisStep={i + 1}
          value={step.value}
        />
      ))}
    </div>
  );
};

export default Stepper;
