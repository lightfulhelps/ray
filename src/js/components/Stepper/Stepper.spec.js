import React from 'react';
import { shallow } from 'enzyme';
import merge from 'lodash/merge';
import Stepper from './Stepper';

const setup = (overrides = {}) => {
  const props = merge(
    {
      children: 'Click Me',
      onClick: jest.fn(),
    },
    overrides
  );
  const wrapper = shallow(<Stepper {...props} />);

  return {
    wrapper,
    props,
  };
};

describe('<Stepper />', () => {
  it('should render', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
