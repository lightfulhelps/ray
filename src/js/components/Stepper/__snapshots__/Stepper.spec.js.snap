// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stepper /> should render 1`] = `
<div
  className="stepper d-flex flex-column flex-sm-row justify-content-between"
>
  <Step
    activeStep={1}
    isLast={false}
    key="0"
    label="step one"
    thisStep={1}
    value={1}
  />
  <Step
    activeStep={1}
    isLast={false}
    key="1"
    label="step two"
    thisStep={2}
    value={2}
  />
  <Step
    activeStep={1}
    isLast={true}
    key="2"
    label="step three"
    thisStep={3}
    value={3}
  />
</div>
`;
